rubbery-raccoons/wired/author_tools/views.py

```diff
@@ -0,0 +1,68 @@
+from datetime import date
+
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.db import IntegrityError
+from django.shortcuts import get_object_or_404, render
+from django.utils.decorators import method_decorator
+from django.utils.text import slugify
+from django.views.generic.list import ListView
+
+from .forms import ArticleForm
+from wired_app.models import Article
+
+
+@login_required
+def compose(request):
+    if request.method == "POST":
+        form = ArticleForm(request.POST)
+        if form.is_valid():
+            art = form.save(commit=False)
+            art.author = request.user
+            art.publication_date = date.today()
+            art.slug = "-".join(
+                (slugify(x) for x in [art.author.username, art.title, art.publication_date])
+            )
+            try:
+                art.save()
+                messages.success(request, "Article submitted!")
+                form = ArticleForm()
+            except IntegrityError:
+                messages.error(request, "Invalid article slug, try changing the title")
+        else:
+            messages.error(request, "Form input was invalid.")
```

Comment:
I would have liked to see a more specific message here describing the error. For example if the user passed an invalid category, it should say something to that effect.